cmake/EnableCompilerWarnings.cmake

```diff
@@ -66,7 +66,7 @@ else()
             -Wunreachable-code
             -Wunused
 
-            -Wno-sign-compare
+            -Wsign-compare
```

Comment:
In what scenario does this checking happen, during "make all" or "make tidy"?

Reply:
It is on for any regular build, not just tidy. Testing was simple - turned it on (removed "no" from sign-compare flag) and saw the flood of warnings until all the issues were fixed.

Reply:
Just want to make sure: if there is a warning, the build will error out. Is that right?